pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/FieldSchemaBuilderImpl.java

```diff
@@ -132,6 +141,10 @@ Field build() {
             case TIMESTAMP:
                 baseSchema = LogicalTypes.timestampMillis().addToSchema(Schema.create(Schema.Type.LONG));
                 break;
+            case AVRO:
+                GenericAvroSchema genericAvroSchema = (GenericAvroSchema) genericSchema;
```

Comment:
You need to check if the `GenericSchema` is AVRO or not before casting it to `GenericAvroSchema`.

```
checkArgument(genericSchema.getSchemaInfo().getSchemaType() == SchemaType.AVRO,
    "The field is expected to be using AVRO schema but " + genericSchema.getSchemaInfo().getSchemaType() + " schema is found");
GenericAvroSchema genericAvroSchema = (GenericAvroSchema) genericSchema;
```